drivers/gles3/rasterizer_canvas_gles3.h

```diff
@@ -156,42 +173,71 @@ class RasterizerCanvasGLES3 : public RendererCanvasRender {
 		GLuint canvas_quad_vertices;
 		GLuint canvas_quad_array;
 
+		GLuint indexed_quad_buffer;
+		GLuint indexed_quad_array;
+
 		GLuint particle_quad_vertices;
 		GLuint particle_quad_array;
 
 		GLuint ninepatch_vertices;
 		GLuint ninepatch_elements;
+
+		RID canvas_shader_default_version;
+
+		uint32_t max_lights_per_render;
+		uint32_t max_lights_per_item;
+		uint32_t max_instances_per_batch = 512;
+		uint32_t max_instances_per_ubo = 16384;
+		uint32_t max_instance_buffer_size = 16384 * 128;
 	} data;
 
+	struct Batch {
+		// Position in the UBO measured in bytes
+		uint32_t start = 0;
+		uint32_t instance_count = 0;
+
+		RID tex = RID();
```

Comment:
Does RID() self initialize in 4.x?

Reply:
Yeah so the point is that usually we wouldn't explicitly initialize it in the header, like `Vector3` or `String`. But it's not a big deal :)